org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/WatchdogCombinedTest.java

```diff
@@ -72,12 +74,31 @@ void createWatchdogs() throws Exception {
 		model.setTopupTime(150);
 		model.setPeriod(5000);
 		
-		this.tdog = new TopupWatchdog(model);
+		tdog = new TopupWatchdog(model);
 		tdog.activate();
 
 	}
 
 	
+	@Before
+	public void before() throws Exception {
+		
+		final IScannable<Number>   topups  = connector.getScannable("topup");
+		final MockTopupScannable   topup   = (MockTopupScannable)topups;
+		assertNotNull(topup);
+		topup.disconnect();
+		Thread.sleep(120); // Make sure it stops, it sets value every 100ms but it should get interrupted

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2925)